### PR TITLE
Update Marina - LDK v0.3.7

### LINKED ISSUE #176: In Assets list do not hide the row when balance 0, but with previous txs

### LINKED ISSUE #186: The user should be able to "deep restore" its wallet
Use the `gapLimit`' restorer parameter implemented in LDK 0.3.2: https://github.com/vulpemventures/ldk/pull/60

### LINKED ISSUE #187: Change network actions should be done in background
...via web-ext redux aliases

### LINKED ISSUE #188: "Undefined" Bug

### PR DESCRIPTION
**Pull request by louisinger**
**This PR updates the code according to the latest LDK changes, refactors some components, and optimizes the way we persist state.**

- LDK 0.3.3 mainly modifies the way we restore identities. That's why now in wallet state, we have the **last indexes of addresses**.
- Now we restore from indexes, thus we can remove the confidential addresses from the state.
- The transactions are now stored as `TxDisplayInterface`. Because we have never used the large `TxInterface` objects stored in state. Replacing them with `TxDisplayInterface` lets to save disk space and a selector in `transactions.tsx` components (so a faster rendering).
- Refactor the `Select` component.
- add the `Settings/DeepRestorer` page (and the deepRestorer backend logic).

it closes #186 
it closes #176 
it closes #187 
it closes #188 

@tiero @Janaka-Steph please review

### CODE REVIEW

**Comment by tiero** [comment]
Stuck here after restore 

image

File: src/application/redux/reducers/wallet-reducer.ts [hunk 1]
```diff
@@ -25,16 +33,27 @@ export function walletReducer(
...
 
-    case ACTION_TYPES.WALLET_SET_ADDRESS_SUCCESS: {
+    case ACTION_TYPES.NEW_CHANGE_ADDRESS_SUCCESS: {
       return {
         ...state,
-        errors: undefined,
-        confidentialAddresses: state.confidentialAddresses.concat([payload.address]),
+        restorerOpts: {
+          ...state.restorerOpts,
+          lastUsedInternalIndex: state.restorerOpts.lastUsedInternalIndex + 1,
```
Comment (Janaka-Steph) [inline src/application/redux/reducers/wallet-reducer.ts:45]: If you set it at "+ 1" it is not the last one anymore

File: src/domain/mnemonic.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
   // Trim start-end and replace multiple spaces in between with a single space
   const mnemonic = mnemo.trim().replace(/ +(?= )/g, '');
 
   try {
     // Mnemonic validation, network doesn't matter
-    new MnemonicTdex({
-      // TODO rename
-      chain: 'liquid',
+    new MnemonicIdentity({
+      chain: network,
```
Comment (Janaka-Steph) [inline src/domain/mnemonic.ts:13]: Does the network matters now? If not we can leave it to 'liquid'

**Comment by tiero** [comment]
> Stuck here after restore
> 
> image

Deleting the cache and reinstalled clean worked again. Noticed is a bit slower than before

File: src/application/redux/reducers/wallet-reducer.ts (hunk 1 above)
Comment (louisinger) [reply src/application/redux/reducers/wallet-reducer.ts:45]: NEW_CHANGE_ADDRESS_SUCCESS is dispatched each time we have generated the `nextChangeAddress`, that's why "+1". What do u mean :thinking: ?

File: src/domain/mnemonic.ts (hunk 2 above)
Comment (louisinger) [reply src/domain/mnemonic.ts:13]: I think we should drop the `new MnemonicIdentity` test and replace it by `bip32.validateMnemonic` function

**Comment by louisinger** [comment]
> > Stuck here after restore
> > image
> 
> Deleting the cache and reinstalled clean worked again. Noticed is a bit slower than before

Yeah the new restorer algorithm is not really optimized, I think we can improve this at the LDK level.

File: src/domain/mnemonic.ts [hunk 3]
```diff
@@ -1,21 +1,13 @@
...
-    new MnemonicIdentity({
-      chain: network,
-      type: IdentityType.Mnemonic,
-      opts: { mnemonic },
-    } as IdentityOpts<MnemonicOpts>);
-    return mnemonic;
-  } catch (err) {
-    throw new Error(err.message);
-  }
+  // Mnemonic validation, network doesn't matter
```
Comment (Janaka-Steph) [inline src/domain/mnemonic.ts:10]: You can remove the comment

File: src/presentation/wallet/transactions/index.tsx [hunk 4]
```diff
@@ -123,7 +129,11 @@ const TransactionsView: React.FC<TransactionsProps> = ({ assets, transactions, n
             alt="liquid bitcoin logo"
           />
           <p className="text-base font-medium">{txTypeAsString(modalTxDetails?.type)}</p>
-          <p className="text-xs font-light">{modalTxDetails?.blockTime?.format('DD MMMM YYYY')}</p>
+          {modalTxDetails && modalTxDetails.blockTimeMs && (
+            <p className="text-xs font-light">
+              {moment(modalTxDetails.blockTimeMs).format('DD MMMM YYYY')}
```
Comment (tiero) [inline src/presentation/wallet/transactions/index.tsx:134]: why not `?`

File: src/domain/mnemonic.ts [hunk 5]
```diff
@@ -1,21 +1,12 @@
-import { IdentityOpts, IdentityType, Mnemonic as MnemonicTdex } from 'ldk';
+import { Network } from './network';
+import { validateMnemonic } from 'bip39';
 
 export type Mnemonic = string;
 
-export function createMnemonic(mnemo: string): Mnemonic {
+export function createMnemonic(mnemo: string, network: Network): Mnemonic {
```
Comment (tiero) [inline src/domain/mnemonic.ts:6]: Is it network needed?

**Comment by tiero** [comment]
some nitpicks, after that LGTM

**Pull request merged**